common/chains/client/rpc_client.go

```diff
@@ -0,0 +1,83 @@
+package client
+
+import (
+	"context"
+	"math/big"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+)
+
+type RPCClient[
+	CHAINID types.ID,
```

Comment:
A few nits.
At other places in txmgr, we have used the generics types as:
CHAIN_ID
BLOCK_HASH
TX_HASH

Could we use the same namings, just to be consistent?
Also same pattern. So EVENTOPS to EVENT_OPS, and TXRECEIPT to TX_RECEIPT

Reply:
done!